api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -352,6 +352,44 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
     return ResponseEntity.ok(new EmptyResponse());
   }
 
+  @Override
+  public ResponseEntity<DataSet> updateDataSet(
+      String workspaceNamespace,
+      String workspaceId,
+      Long dataSetId,
+      DataSetRequest request) {
+    org.pmiops.workbench.db.model.DataSet dbDataSet =
+        getDbDataSet(workspaceNamespace, workspaceId, dataSetId, WorkspaceAccessLevel.WRITER);
+    if(Strings.isNullOrEmpty(request.getEtag())) {
+      throw new BadRequestException("missing required update field 'etag'");
+    }
+    int version = Etags.toVersion(request.getEtag());
+    if (dbDataSet.getVersion() != version) {
+      throw new ConflictException("Attempted to modify outdated data set version");
+    }
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+    dbDataSet.setLastModifiedTime(now);
```

Comment:
I likes the way how attributes are set in data set in **createDataSet** ( eg . DataSet result = new DataSet()
              .name(dataSet.getName())
              .includesAllParticipants(dataSet.getIncludesAllParticipants())....)
Can this be done in the same way

Reply:
That only works on chaining off of a constructor. It would theoretically be possible here, but we'd need to add a wrapper method around set that does things the way we want, which would require some DB modeling refactoring.